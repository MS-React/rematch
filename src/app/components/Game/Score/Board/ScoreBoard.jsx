import React from 'react';
import PropTypes from 'prop-types';

const ScoreBoard = ({
  success, fails, score, resetGame,
}) => (
  <div key="score-user-board" className="score-board">
    <h2>Score Board </h2>
    <h3>Hey! good match</h3>
    <div className="user-report">
      Successful : {success} <br />
      Fails : {fails} <br />
      Total Score : {score} <br />
      <button type="submit" onClick={resetGame}>Try Again! </button>
    </div>
  </div>
);

ScoreBoard.propTypes = {
  success: PropTypes.number,
  fails: PropTypes.number,
  score: PropTypes.number,
  resetGame: PropTypes.func.isRequired,
};

ScoreBoard.defaultProps = {
  success: 0,
  fails: 0,
  score: 0,
};

export default ScoreBoard;
