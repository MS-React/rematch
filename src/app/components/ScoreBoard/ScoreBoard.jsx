import React from 'react';

const ScoreBoard = ({ success, fails, score }) => {
  return (
    <div className="score-board">
      <h2>Score Board</h2>
      <h3>Hey! good match</h3>
      <div>
        Successful : {success} <br />
        Fails : {fails} <br />
        Total Score : {score} <br />
        <button>Try Again!</button>
      </div>
    </div>
  );
}

export default ScoreBoard;
