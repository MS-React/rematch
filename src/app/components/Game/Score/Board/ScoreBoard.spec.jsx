import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ScoreBoard from './ScoreBoard';

describe('ScoreBoard', () => {
  let wrapper;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      success: 0,
      fails: 0,
      score: 0,
      resetGame: jest.fn()
    };
    wrapper = shallow(<ScoreBoard {...defaultProps} />);
  });

  it('should show user result report and a button to restart the game', () => {
    expect(renderer.create(wrapper).toJSON())
      .toMatchSnapshot();
  });
});
