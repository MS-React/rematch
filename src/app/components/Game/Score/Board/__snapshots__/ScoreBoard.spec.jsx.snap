// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard should show user result report and a button to restart the game 1`] = `
<div
  className="score-board"
>
  <h2>
    Score Board 
  </h2>
  <h3>
    Hey! good match
  </h3>
  <div
    className="user-report"
  >
    Successful : 
    0
     
    <br />
    Fails : 
    0
     
    <br />
    Total Score : 
    0
     
    <br />
    <button
      onClick={[MockFunction]}
      type="submit"
    >
      Try Again! 
    </button>
  </div>
</div>
`;
